core/src/crypto/BLAKE.cpp

```diff
@@ -54,6 +54,15 @@ namespace ledger {
             return hash;
         }
 
+        std::vector<uint8_t> BLAKE::blake2b(const std::vector<uint8_t>& data, size_t outLength) {
```

Comment:
I suggest to change the interface here to make it less coupled on the container. The internal representation only needs a buffer, so imposing vector is a bit too strong. See my other comment on the use of this function.

Reply:
Iterators would be C++ way to do this. But it will differ from all other functions in this file that accept vector<uint8_t>, imo this should be separate commit that change vector<> to iterators, if we eventually want this.

Reply:
Problems with iterators is that they do not ensure contiguity (unless we use a `ContiguousIterator` concept?).

Reply:
+1 @teams2ua , that's why the method has this signature 
@phaazon don't forget that this methods are used internally only with byte arrays